alea/parameters.py

```diff
@@ -94,6 +100,26 @@ def fit_guess(self) -> float:
     def fit_guess(self, value: float) -> None:
         self._fit_guess = value
 
+    @property
+    def parameter_interval_bounds(self) -> float:
+        # make sure to only return parameter_interval_bounds if fittable
+        if self._parameter_interval_bounds is not None and not self.fittable:
+            raise ValueError(
+                f"Parameter {self.name} is not fittable, but has a parameter_interval_bounds.")
+        else:
+            return self._parameter_interval_bounds
+
+    @parameter_interval_bounds.setter
+    def parameter_interval_bounds(self, value: Optional[List]) -> None:
+        if value is None:
+            value = [-MAX_FLOAT, MAX_FLOAT]
+        else:
+            if value[0] is None:
+                value[0] = -MAX_FLOAT
```

Comment:
If set the boundary `` to np.inf, we will see errors like:

```
  0%|          | 0/3 [00:11<?, ?it/s]
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[31], line 1
----> 1 toydata, results = runner.toy_simulation()
      2 runner.write_output(results)

File ~/alea/alea/runner.py:229, in Runner.toy_simulation(self)
    227 fit_result['ll'] = max_llh
    228 if self._compute_confidence_interval and (self.poi not in hypothesis_values):
--> 229     dl, ul = self.statistical_model.confidence_interval(
    230         poi_name=self.poi,
    231         best_fit_args=self._hypotheses_values[0],
    232         confidence_interval_args=hypothesis_values)
    233 else:
    234     dl, ul = np.nan, np.nan

File ~/alea/alea/model.py:465, in StatisticalModel.confidence_interval(self, poi_name, parameter_interval_bounds, confidence_level, confidence_interval_kind, best_fit_args, confidence_interval_args)
    463 if confidence_interval_kind in {"upper", "central"} and t_best_parameter < 0:
    464     if t(parameter_interval_bounds[1]) > 0:
--> 465         ul = brentq(t, best_parameter, parameter_interval_bounds[1])
    466     else:
    467         ul = np.inf

File /opt/XENONnT/anaconda/envs/XENONnT_2023.07.2/lib/python3.9/site-packages/scipy/optimize/_zeros_py.py:784, in brentq(f, a, b, args, xtol, rtol, maxiter, full_output, disp)
    782 if rtol < _rtol:
    783     raise ValueError("rtol too small (%g < %g)" % (rtol, _rtol))
--> 784 r = _zeros._brentq(f, a, b, xtol, rtol, maxiter, args, full_output, disp)
    785 return results_c(full_output, r)

RuntimeError: Failed to converge after 100 iterations.
```

This is because `scipy.optimize.brentq` can not accept `np.inf` as the boundary. Slack discussion: https://xenonnt.slack.com/archives/C04JRF0AZTP/p1690870295199999

Reply:
Which will make the brentq super slow-- I think we should rather ask for a sensible range to always be given.

Reply:
Maybe give a warning?

Reply:
Well, even if I change `MAX_FLOAT` from 10^19 to 100, the intention time did not change. 

Checked by
```
ul, r = brentq(t, best_parameter, parameter_interval_bounds[1], full_output=True)
print(r)
```